.github/workflows/promote.yml

```diff
@@ -0,0 +1,64 @@
+name: promote
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+      module:
+        required: true
+        type: string
+  pull_request:  
+  workflow_dispatch:
+    inputs:
+      environment:
+        required: true
+        type: choice
+        options:
+          - dev
+          - test
+          - sbx
+          - prod
+          - prod-test  
+  
+jobs:
+  promote:
+    runs-on: self-hosted
```

Comment:
This can run on ubuntu-latest since it doesn't rely on anything in our internal network.